ecr-login/ecr.go

```diff
@@ -70,14 +71,38 @@ func NewECRHelper(opts ...Option) *ECRHelper {
 // ensure ECRHelper adheres to the credentials.Helper interface
 var _ credentials.Helper = (*ECRHelper)(nil)
 
-func (ECRHelper) Add(creds *credentials.Credentials) error {
-	// This does not seem to get called
-	return notImplemented
+func shouldIgnoreCredsStorage() bool {
+	return os.Getenv("AWS_ECR_IGNORE_CREDS_STORAGE") == "true"
 }
 
-func (ECRHelper) Delete(serverURL string) error {
-	// This does not seem to get called
-	return notImplemented
+// Add tries to store credentials when docker requests it. This usually happens during `docker login` calls. In our context,
+// storing arbitrary user given credentials makes no sense.
+func (self ECRHelper) Add(creds *credentials.Credentials) error {
+	if shouldIgnoreCredsStorage() {
+		self.logger.
+			WithField("serverURL", creds.ServerURL).
+			Warning("Ignoring request to store credentials since AWS_ECR_IGNORE_CREDS_STORAGE env variable is set." +
+				"ecr-login does not require 'docker login', and does not support persisting temporary ECR-issued credentials.")
+		return nil
+	} else {
+		self.logger.Warning("Saving credentials is not supported by the ecr-login credentials helper as all issued credentials are temporary. Consider setting the AWS_ECR_IGNORE_CREDS_STORAGE env variable (see documentation for details).")
```

Comment:
I prefer to be more explicit here. Technically, ecr-login is already saving credentials through its file cacher.
```suggestion
		self.logger.Warning("Add() is not supported by the ecr-login credentials helper. Consider setting the AWS_ECR_IGNORE_CREDS_STORAGE env variable (see documentation for details)")
```

Reply:
Fair enough. This, and the similar comment below has been added!